experimental/unitsconversion/src/power.rs

```diff
@@ -2,27 +2,47 @@
 // called LICENSE at the top level of the ICU4X source tree
 // (online at: https://github.com/unicode-org/icu4x/blob/main/LICENSE ).
 
+use zerotrie::{ZeroTrieSimpleAscii, ZeroTrieSimpleAsciiCursor};
+
 // TODO: consider returning Option<(u8, &str)> instead of (1, part) for the case when the power is not found.
 // TODO: complete all the cases for the powers.
 // TODO: consider using a trie for the powers.
-/// Converts a power string to a power.
-pub fn get_power(part: &str) -> Option<i8> {
-    match part {
-        "pow1" => Some(1),
-        "square" | "pow2" => Some(2),
-        "cubic" | "pow3" => Some(3),
-        "pow4" => Some(4),
-        "pow5" => Some(5),
-        "pow6" => Some(6),
-        "pow7" => Some(7),
-        "pow8" => Some(8),
-        "pow9" => Some(9),
-        "pow10" => Some(10),
-        "pow11" => Some(11),
-        "pow12" => Some(12),
-        "pow13" => Some(13),
-        "pow14" => Some(14),
-        "pow15" => Some(15),
-        _ => None,
+/// Extracts the power from the given CLDR ID part.
+///     - If the power is not found, the function returns (1, part).
+///     - If the power is found, the function will return (power, part without the string of the power).
+pub fn get_power(part: &str) -> (u8, &str) {
+    use std::collections::BTreeMap;
+    let mut powers = BTreeMap::<Vec<u8>, usize>::new();
+    powers.insert(b"pow1".to_vec(), 1);
+    powers.insert(b"pow2".to_vec(), 2);
+    powers.insert(b"square".to_vec(), 2);
+    powers.insert(b"pow3".to_vec(), 3);
+    powers.insert(b"cubic".to_vec(), 3);
+    powers.insert(b"pow4".to_vec(), 4);
+    powers.insert(b"pow5".to_vec(), 5);
+    powers.insert(b"pow6".to_vec(), 6);
+    powers.insert(b"pow7".to_vec(), 7);
+    powers.insert(b"pow8".to_vec(), 8);
+    powers.insert(b"pow9".to_vec(), 9);
+    powers.insert(b"pow10".to_vec(), 10);
+    powers.insert(b"pow11".to_vec(), 11);
+    powers.insert(b"pow12".to_vec(), 12);
+    powers.insert(b"pow13".to_vec(), 13);
+    powers.insert(b"pow14".to_vec(), 14);
+    powers.insert(b"pow15".to_vec(), 15);
+
+    let trie = ZeroTrieSimpleAscii::try_from(&powers).unwrap();
```

Comment:
@sffc How to construct the try directly  without the need of an intermediate map?

Also, the current way, will construct the map at each function call, how to build it at the compile time?

Reply:
Shall I use `lazy_static`?

Reply:
No, use the const constructor of ZeroTrieSimpleAscii

https://unicode-org.github.io/icu4x/rustdoc/zerotrie/struct.ZeroTrieSimpleAscii.html#method.from_sorted_str_tuples

You need to manually put the strings in the correct order and then select the correct trie length. The error message should tell you whether the length is too long or too short and then you bisect to find the correct length. Alternatively you can use the non-const constructor to find the length.

Reply:
I made it easier in https://github.com/unicode-org/icu4x/pull/4466

Reply:
done, thanks for the info

I would like if we can give it unsorted.